lang/ast/src/exp/hole.rs

```diff
@@ -46,6 +49,24 @@ pub struct Hole {
     pub solution: Option<Box<Exp>>,
 }
 
+impl Hole {
+    /// The context of the hole arguments
+    pub fn levels(&self) -> LevelCtx {
+        let bound = self
+            .args
+            .iter()
+            .map(|args| {
+                args.iter()
+                    // FIXME: Track the variable names of hole arguments
```

Comment:
I think in order to fix this we should change the definition of the `args` field of Holes to `GenericCtx<Box<Exp>>`. (This is isomorphic to typing contexts, but we use the `Exp` for the arguments.